src/DataStreams/PushingToViewsBlockOutputStream.cpp

```diff
@@ -334,6 +341,19 @@ void PushingToViewsBlockOutputStream::process(const Block & block, ViewInfo & vi
 {
     Stopwatch watch;
 
+    auto * original_thread = current_thread;
+    SCOPE_EXIT({ current_thread = original_thread; });
+
+    if (view.runtime_stats.thread_status)
+    {
+        /// Change thread context to store individual metrics per view. Once the work in done, go back to the original thread
+        *view.runtime_stats.thread_status->last_rusage = RUsageCounters::current();
```

Comment:
What is RUsageCounters? What is it responsible for?

Reply:
[Real, user and system time and page faults](https://github.com/ClickHouse/ClickHouse/blob/5a2bd0b6bad1577e48fd00571bd376befbf70cb9/src/Common/ThreadProfileEvents.h#L78). These are part of the profile events.